indexing-service/src/main/java/org/apache/druid/indexing/common/task/batch/parallel/ParallelIndexSupervisorTask.java

```diff
@@ -360,43 +362,77 @@ SegmentIdWithShardSpec allocateNewSegment(DateTime timestamp) throws IOException
   {
     final String dataSource = getDataSource();
     final GranularitySpec granularitySpec = getIngestionSchema().getDataSchema().getGranularitySpec();
-    final SortedSet<Interval> bucketIntervals = Preconditions.checkNotNull(
-        granularitySpec.bucketIntervals().orNull(),
-        "bucketIntervals"
-    );
+    final Optional<SortedSet<Interval>> bucketIntervals = granularitySpec.bucketIntervals();
+
     // List locks whenever allocating a new segment because locks might be revoked and no longer valid.
     final Map<Interval, String> versions = toolbox
         .getTaskActionClient()
         .submit(new LockListAction())
         .stream()
+        // FIXME note that the next line is new in this commit --- it's not relevant for fixing #6989,
+        //       it just seems like a bug fix to make the code match its comment
```

Comment:
Thanks for trying to fix this! I think it's not a bug, but it looks useful because it would make the task failed earlier if it doesn't a valid lock.

The intention of returning revoked locks in LockListAction is that the caller should be notified about revoked locks, so that they can handle it properly. I think we should fail here immediately if any lock is revoked because there's no way to get a new lock once the lock is revoked.

Reply:
The reason I think the current code doesn't match is comment is that if we didn't care about checking for revoked locks, we wouldn't need to run LockListAction again – we could just remember the versions we got when we took the locks out at the beginning of the task.